app/models/invoice.rb

```diff
@@ -106,20 +106,66 @@ def subscription_fees(subscription_id)
   end
 
   def recurring_fees(subscription_id)
-    subscription_fees(subscription_id).joins(charge: :billable_metric)
-      .merge(BillableMetric.recurring_count_agg)
+    subscription_fees(subscription_id)
+      .joins(charge: :billable_metric)
+      .where(billable_metric: { recurring: true })
+      .where(billable_metric: { aggregation_type: [1, 3] })
+      .where(charge: { pay_in_advance: false })
   end
 
   def recurring_breakdown(fee)
-    result = BillableMetrics::Aggregations::RecurringCountService.new(
+    service = case fee.charge.billable_metric.aggregation_type.to_sym
+              when :sum_agg
+                BillableMetrics::Breakdown::SumService
+              when :unique_count_agg
+                BillableMetrics::Breakdown::UniqueCountService
+              else
+                raise(NotImplementedError)
+              end
+
+    service.new(
       billable_metric: fee.charge.billable_metric,
       subscription: fee.subscription,
       group: fee.group,
     ).breakdown(
       from_datetime: DateTime.parse(fee.properties['charges_from_datetime']),
       to_datetime: DateTime.parse(fee.properties['charges_to_datetime']),
+    ).breakdown
+  end
+
+  def charge_pay_in_advance_proration_range(fee, timestamp)
+    date_service = Subscriptions::DatesService.new_instance(
+      fee.subscription,
+      Time.zone.at(timestamp),
+      current_usage: true,
     )
-    result.breakdown
+
+    event = Event.find_by(id: fee.pay_in_advance_event_id)
+
+    return {} unless event
+
+    number_of_seconds = date_service.charges_to_datetime.in_time_zone(customer.applicable_timezone) -
+                        event.timestamp.in_time_zone(customer.applicable_timezone)
+
+    number_of_days = number_of_seconds.fdiv(86_400).round
```

Comment:
You could use some Rails magic:
```suggestion
    number_of_days = number_of_seconds.fdiv(1.day).round
```